tests/unit/test_crs.py

```diff
@@ -0,0 +1,163 @@
+
+
+def test_parse_gpd_file(tmp_path):
+    """Test GPD files are correctly parsed."""
+    gpd_content = dedent(
+        """
+    ; Example gpd for parsing
+    Map Origin X: -7000.0  ; pinned location for Map Origin X
+    Map Origin Y: 0.0
+    Grid Map Units per Cell: 1000.0
+    Grid Map Origin Column: -0.5
+    Grid Map Origin Row: -0.5
+    Something That Exists But Is Not Used: Has A Value
+    """
+    ).strip()
+
+    gpd_file = tmp_path / "test.gpd"
+    gpd_file.write_text(gpd_content)
+
+    result = parse_gpd_file(str(gpd_file))
+    assert result['Map Origin X'] == -7000.0
+    assert result['Map Origin Y'] == 0.0
+    assert result['Grid Map Units per Cell'] == 1000.0
+    assert result['Something That Exists But Is Not Used'] == 'Has A Value'
+
+
+def test_create_crs():
+    """Test create crs transforms input info into array with metadata."""
+    crs_data_array = create_crs(sample_target_info)
+    assert 'crs_wkt' in crs_data_array.attrs
+    assert (
+        'NSIDC EASE-Grid 2.0 Global' in crs_data_array.crs_wkt
+    ), f'expected "NSIDC EASE-Grid 2.0 Global" in {crs_data_array.crs_wkt}'
+    assert crs_data_array.attrs['proj'] == (
+        '+proj=cea +lat_ts=30 +lon_0=0 +x_0=0 '
+        '+y_0=0 +datum=WGS84 +units=m +no_defs +type=crs'
+    )
+    assert isinstance(crs_data_array, DataArray)
+
+
+def test_compute_dims():
+    """Test compute dims against a known entity.
+
+    Use the EASE2 Global 9km grid. and verify a few values against known values
+    pulled form NSIDC-0772 dataset
+
+    """
+    EASE2_M09km_grid_info = parse_gpd_file(EASE2_M09km_gpd_filename)
+    x_dim, y_dim = compute_dims(EASE2_M09km_grid_info)
+    assert len(x_dim) == EASE2_M09km_grid_info['Grid Width']
+    assert len(y_dim) == EASE2_M09km_grid_info['Grid Height']
+    selected_y_index = np.array([0, 203, 406, 609, 812, 1014, 1217, 1420, 1623])
+    selected_x_index = np.array([0, 482, 964, 1446, 1928, 2409, 2891, 3373, 3855])
+    expected_x_values = np.array(
+        [
+            -1.7363026417556427e07,
+            -1.3021143806266051e07,
+            -8.6792611949756760e06,
+            -4.3373785836853012e06,
+            4.5040276050753891e03,
+            4.3373785836853050e06,
+            8.6792611949756779e06,
+            1.3021143806266055e07,
+            1.7363026417556427e07,
+        ],
+        dtype=np.float64,
+    )
+    expected_y_values = np.array(
+        [
+            7.3100368030335270e06,
+            5.4814015953738764e06,
+            3.6527663877142267e06,
+            1.8241311800545761e06,
+            -4.5040276050735265e03,
+            -1.8241311800545771e06,
+            -3.6527663877142277e06,
+            -5.4814015953738783e06,
+            -7.3100368030335270e06,
+        ],
+        dtype=np.float64,
+    )
+
+    np.testing.assert_array_almost_equal(x_dim[selected_x_index], expected_x_values)
+    np.testing.assert_array_almost_equal(y_dim[selected_y_index], expected_y_values)
```

Comment:
Are we testing almost equal because these are floating point values?

Reply:
Yes exactly that.
```
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 8 / 9 (88.9%)
E           Max absolute difference among violations: 2.23517418e-08
E           Max relative difference among violations: 3.30840805e-12
```